webui/react/src/pages/Templates/TemplateCreateModal.tsx

```diff
@@ -88,27 +120,31 @@ const TemplateCreateModalComponent: React.FC<Props> = ({ workspaceId, onSuccess
         form: idPrefix + FORM_ID,
         handleError,
         handler: handleSubmit,
-        text: 'Create Template',
+        text: `${template ? 'Edit' : 'Create'} Template`,
       }}
-      title="New Template">
+      title={`${template ? 'Edit' : 'New'} Template`}>
       <Form
         autoComplete="off"
         form={form}
         id={idPrefix + FORM_ID}
         layout="vertical"
         onFieldsChange={onChange}>
         <Form.Item
+          initialValue={template?.name}
           label="Name"
           name="name"
           rules={[{ message: 'Name is required.', required: true }]}>
           <Input />
         </Form.Item>
         <Form.Item
-          initialValue={workspaceId}
+          initialValue={template?.workspaceId || workspaceId}
```

Comment:
**nit:** using `??` instead of `||` in case `template?.workspaceId` is `0` as a valid id (not likely but just in case)